telmos_main.py

```diff
@@ -36,56 +24,253 @@ def telmos_main(delta_root, tmfs_root, tel_year, tel_id, tel_scenario,
     suffixes = range(3, 9)
     
     sr_array = []
-    all_names = []
     for suffix in suffixes:
         data = []
         names = []
         for period, purpose, mode in product(periods, purposes, modes):
             file_name = "%s_%s_%s_%s.txt" % (purpose, mode, period, suffix)
-            factor_file = os.path.join(factors_base, file_name)
-            try:
-                data.append(np.loadtxt(factor_file))
-                names.append(file_name)
-            except FileNotFoundError as f:
-                # If any file can not be found - abort
-                log_func("Could not find file %s\nAborting" % f)
-                return
+            factor_file = os.path.join(factors_dir, file_name)
+            data.append(np.loadtxt(factor_file))
+            names.append(file_name)
         sr_array.append(data)
-        all_names.append(names)
-    sr_array = np.asarray(sr_array)
+    return np.asarray(sr_array)
+
+def load_cte_tod_files(tod_files, cte_files, file_base):
+    tod_data = []
+    cte_data = []
+    for t_file, c_file in zip(tod_files, cte_files):
+        # TMfS14 had a long-distance module that required some CTE/TOD files to have
+        # _All appended to the end. This can now be removed if needed.
+        if (os.path.exists(os.path.join(file_base, t_file)) and
+               os.path.exists(os.path.join(file_base, c_file))):
+            t_file_name = t_file
+            c_file_name = c_file
+        else:
+            t_file_name = t_file.replace("_ALL","")
+            c_file_name = c_file.replace("_ALL","")
+        tod_data.append(np.loadtxt(os.path.join(file_base, t_file_name),
+                                   delimiter=","))
+        cte_data.append(np.loadtxt(os.path.join(file_base, c_file_name),
+                                   delimiter=","))
+    return (np.asarray(tod_data), np.asarray(cte_data))
+
+def create_attraction_pivot(planning_data, attraction_trip_rates):
+    ''' 
+    Multiplies the planning data and the attraction trip rates
+    '''
+    # Columns are: Work, Employment, Other, Education
+    attr_factors_array = np.zeros((planning_data.shape[0], 4),dtype="float32")
+    attr_factors_array[:,0] = planning_data[:,2] * attraction_trip_rates[0, 0]
+    attr_factors_array[:,1] = planning_data[:,[1,3,4,5,6,7,8]].sum(axis=1) * attraction_trip_rates[2,1]
+    attr_factors_array[:,2] = (planning_data[:,[4,8,1,3]] * 
+                      attraction_trip_rates[[6,7,1,12],[3,4,8,9]]).sum(axis=1)
+    attr_factors_array[:,3] = planning_data[:,7] * attraction_trip_rates[2,2]
+    # Fill zeros with ones
+    attr_factors_array[attr_factors_array == 0.0] = 1.0
+    return attr_factors_array
```

Comment:
If we're just resetting to 1 anyway, can we not just use `np.ones`?

```suggestion
    attr_factors_array = np.ones((planning_data.shape[0], 4),dtype="float32")
    attr_factors_array[:,0] = planning_data[:,2] * attraction_trip_rates[0, 0]
    attr_factors_array[:,1] = planning_data[:,[1,3,4,5,6,7,8]].sum(axis=1) * attraction_trip_rates[2,1]
    attr_factors_array[:,2] = (planning_data[:,[4,8,1,3]] * 
                      attraction_trip_rates[[6,7,1,12],[3,4,8,9]]).sum(axis=1)
    attr_factors_array[:,3] = planning_data[:,7] * attraction_trip_rates[2,2]
    return attr_factors_array
```